### PR TITLE
Add filter for valid fields. Fixes #4098

### LINKED ISSUE #4098: Low contrast in binned heatmap
See https://vega.github.io/vega-lite/examples/rect_binned_heatmap.html

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/predicate.ts [hunk 1]
```diff
@@ -139,12 +141,20 @@ export interface FieldOneOfPredicate extends FieldPredicateBase {
   oneOf: string[] | number[] | boolean[] | DateTime[];
 }
 
+export interface FieldValidPredicate extends FieldPredicateBase {
+  valid: true;
```
Comment (kanitw) [inline src/predicate.ts]: This should support false too.  Users may want to analyze the pattern of data that have some fields missing.

File: site/docs/transform/filter.md [hunk 2]
```diff
@@ -112,7 +112,16 @@ For example, to check if the `height` field's value is greater than or equals to
...
-For example, `{"filter": {"field": "car_color", "oneOf":["red", "yellow"]}}` checks if the `car_color` field's value is `"red"` or `"yellow"`.
+For example, `{"filter": {"field": "car_color", "oneOf": ["red", "yellow"]}}` checks if the `car_color` field's value is `"red"` or `"yellow"`.
+
+
+{:#valid-predicate}
+### Field Valid Predicate
+
+{% include table.html props="valid" source="FieldValidPredicate" %}
+
+For example, `{"filter": {"field": "car_color", "valid": true}}` checks if the `car_color` field's value is not `null` and whether it is not `NaN`.
```
Comment (kanitw) [inline site/docs/transform/filter.md]: Adjust the language to make it clear which data items would be included.

File: src/predicate.ts (hunk 1 above)
Comment (domoritz) [reply src/predicate.ts]: You can always use `not`. I'd rather support fewer ways to do the same thing for now. We can change it later.

**Comment by domoritz** [comment]
Done

File: src/predicate.ts (hunk 1 above)
Comment (kanitw) [reply src/predicate.ts]: I disagree.  Users wouldn't expect that it can't be `false`, so better fix it now since it's very easy and provides a way nicer syntax that nesting it under "not".

**Comment by domoritz** [comment]
I added some tests

File: src/predicate.ts [hunk 3]
```diff
@@ -216,7 +229,9 @@ export function fieldFilterExpression(predicate: FieldPredicate, useInRange = tr
     const lower = predicate.gte;
     return `${fieldExpr}>=${predicateValueExpr(lower, timeUnit)}`;
   } else if (isFieldOneOfPredicate(predicate)) {
-    return 'indexof([' + predicateValuesExpr(predicate.oneOf, timeUnit).join(',') + '], ' + fieldExpr + ') !== -1';
+    return `indexof([${predicateValuesExpr(predicate.oneOf, timeUnit).join(',')}], ${fieldExpr}) !== -1`;
+  } else if (isFieldValidPredicate(predicate)) {
+    return predicate.valid ? `${fieldExpr}!==null&&!isNaN(${fieldExpr})` : `${fieldExpr}===null||isNaN(${fieldExpr})`;
```
Comment (kanitw) [inline src/predicate.ts]: I'd prefer having spaces for readability.  

It might be worth submitting another PR to standardize other the filter expressions too.  
(Currently, some do have spaces between operators, some do not.)

Comment (domoritz) [reply src/predicate.ts]: Let's do this in a follow up. Issue: https://github.com/vega/vega-lite/issues/4112
